core/domain/feedback_domain.py

```diff
@@ -107,6 +109,37 @@ def get_thread_id_from_full_thread_id(full_thread_id):
         """
         return full_thread_id.split('.')[1]
 
+    def get_full_message_id(self, message_id):
+        """ Returns the full id of the message.
+
+        Args:
+            message_id: int. The id of the message for which we have to fetch
+                the complete message id.
+
+        Returns:
+            str. The full id corresponding to the given message id.
+        """
+        return '.'.join([self.id, str(message_id)])
+
+    def get_last_two_message_ids(self):
+        """Returns the full message ids of the last two messages of the thread.
+
+        Returns:
+            list(str). The ids of the last two messages of the thread.
+        """
+        message_ids = []
+        last_message_id = self.message_count - 1
+        if self.message_count > 1:
+            second_last_message_id = self.message_count - 2
+        else:
+            # The second last message does not exist. So we set it's id to -1
+            # implying that it doesn't exist.
+            second_last_message_id = -1
```

Comment:
No, this is really iffy. You shouldn't be constructing full message IDs from message IDs that are invalid. At this point, if you are appending None, then just do so here rather than exposing the "badness" to other methods.